### PR TITLE
Butovens_Médé_lab_4_completed

### PR DESCRIPTION
**Pull request by Butovens**

### CODE REVIEW

**Comment by Butovens** [comment]
For some reason the testthat::test_dir('tests') never works on my laptop. It just stops with that error message 
"Error: C stack usage  7971472 is too close to the limit" after giving my a bunch of "✓ |  OK F W S | Context"

File: lab.R [hunk 1]
```diff
@@ -21,50 +21,51 @@
...
 library(visdat)
-
+library(testthat)
 library(AmesHousing)
 
 dataf = read_csv(file.path('data', 'ames.csv'))
 
 ## To check your answers locally, run the following: 
-## testthat::test_dir('tests')
+testthat::test_dir('tests')
```
Comment (dhicks) [inline lab.R]: This is causing an infinite loop:  The first thing the test file does is run your script, and this line runs the test file.  So leave this commented out in your script.  I've definitely realized that, next time around, I need to explain this infrastructure more carefully for the first lab. 
```suggestion
# testthat::test_dir('tests')
```

File: lab.R [hunk 2]
```diff
@@ -139,22 +147,23 @@ levels(bad_factor)
...
 #' Using `mutate()`, add a variable `overall_cond_fct` to `dataf`, that represents overall condition as a factor with the levels in the correct order. 
 
-# dataf = ???
+dataf = mutate(dataf,overall_cond_fact = good_factor)
 
 
 #' # Problem 6 #
 #' Recall that we're interested in finding variables that are highly correlated with sale price.  We can use the function `cor()` to construct a correlation matrix, with correlations between all pairs of variables in the dataframe.  But this creates two challenges.  First, `cor()` only works with numerical inputs.  If we try it with our current dataframe, it throws an error:  
-# cor(dataf)
+cor(dataf)
```
Comment (dhicks) [inline lab.R]: This is here so you can see that it throws an error.  But we don't want to throw the error. 
```suggestion
# cor(dataf)
```

**Review by dhicks — CHANGES_REQUESTED** [review]
On my machine, the script now runs correctly.  It looks like there are still some things you need to finish in problems 5 and 6.

File: lab.R (hunk 1 above)
Comment (Butovens) [reply lab.R]: Oh  see! Thank you Dan, that solved it

File: lab.R (hunk 2 above)
Comment (Butovens) [reply lab.R]: My mistake, I uncommented it to see to output and forgot to comment it out back.

**Comment by Butovens** [comment]
The mistake that caused my checks not to pass for Q.5 and 6 was that I unintentionally renamed one of the variables (i.e. "overall_cond_fact" instead of "overall_cond_fct" ). Thus, even if the coding was accurate the answer was not recognized as such. Thanks again!

**Comment by dhicks** [comment]
Looks good, thanks!

**Pull request closed without merging**